R/validate-policy.R

```diff
@@ -1,3 +1,45 @@
+#' Validate Policy File
+#' 
+#' Flattens and validates policy file
+#'  @param policy YAML policy file
+#'  @param gs Gradescope data
+#'  @importFrom purrr map discard
+#'  @export
+validate_policy <- function(policy, gs){
+  policy <- flatten_policy(policy)
+  # drop categories with unavailable assignments
+  categories <- map(policy$categories, "category") |> unlist()
+  assignments <- c(get_assignments(gs), categories)
+  policy$categories <- map(policy$categories, function(cat){
+    cat$assignments <- cat$assignments[cat$assignments %in% assignments]
+    if (length(cat$assignments) == 0){
+      return (NULL)
+    }
+    return (cat)
+  }) |>
+    purrr::discard(is.null)
+  
+  default_cat <- list(
+    aggregation = "equally_weighted",
+    aggregation_max_pts = "sum_max_pts",
+    aggregation_lateness = "max_lateness"
+  ) 
+  
+  # add default values if missing
+  policy$categories <- map(policy$categories, function(cat){
```

Comment:
Very nice way to do this. With `default_cat` stored as a separate list, it'll be easy to expand/update this as needed.